lib/rubocop/cop/rails/env.rb

```diff
@@ -0,0 +1,62 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Rails
+      # This cop checks typo of environment name.
+      #
+      # @example
+      #   # bad
+      #   Rails.env.proudction?
+      #
+      #   # good
+      #   Rails.env.production?
+      class Env < Cop
```

Comment:
Would probably want a more descriptive name to make it easier to tell what it does, and to not clash with other potential cops. E.g. might want a cop that checks that `Rails.env` is used instead of `ENV["RAILS_ENV"]`.